book-service/src/main/java/com/jpop8/bookservice/resources/BookResource.java

```diff
@@ -0,0 +1,62 @@
+package com.jpop8.bookservice.resources;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.jpop8.bookservice.exceptions.ResourceNotFoundException;
+import com.jpop8.bookservice.models.Book;
+import com.jpop8.bookservice.services.BookService;
+
+import lombok.extern.slf4j.Slf4j;
+
+@CrossOrigin(origins = "http://localhost:4200")
+@Slf4j
+@RestController
+@RequestMapping("/books")
+public class BookResource {
+
+	private BookService bookService;
+
+	public BookResource(BookService bookService) {
+		this.bookService = bookService;
+	}
+
+	@GetMapping
+	public ResponseEntity<List<Book>> getAllBooks(@RequestParam(required = false) String name) {
+		log.debug("Going to fetch books");
+		return new ResponseEntity<>(bookService.getAllBooks(name), HttpStatus.OK);
+	}
+
+	@GetMapping("/{id}")
```

Comment:
Try to follow versioning.
https://dzone.com/articles/versioning-rest-api-with-spring-boot-and-swagger